vllm/v1/engine/output_processor.py

```diff
@@ -74,6 +85,94 @@ def from_new_request(
             log_stats=log_stats,
         )
 
+    def make_request_output(
+        self,
+        new_token_ids: List[int],
+        finish_reason: Optional[FinishReason],
+        stop_reason: Union[int, str, None],
+    ) -> Optional[RequestOutput]:
+
+        finished = finish_reason is not None
+        output_kind = self.output_kind
+
+        # In follow up, we will switch to invariant where EngineCore
+        # does not stream partial prefills.
+        if not finished and (self.is_prefilling
+                             or output_kind == RequestOutputKind.FINAL_ONLY):
```

Comment:
Have an `final_only` var that can be used here and below?

Reply:
Sounds good :+1: